EpiAware/src/EpiObsModels/StackObservationModels.jl

```diff
@@ -0,0 +1,112 @@
+@doc raw"
+
+A stack of observation models that are looped over to generate observations for
+each model in the stack. Note that the model names are used to prefix the parameters in each model (so if I have a model named `cases` and a parameter `y_t`, the parameter in the model will be `cases.y_t`).
+
+## Constructors
+
+- `StackObservationModels(models::Vector{<:AbstractTuringObservationModel},
+    model_names::Vector{<:AbstractString})`: Construct a `StackObservationModels` object with a vector of observation models and a vector of model names.
+    - `StackObservationModels(; models::Vector{<:AbstractTuringObservationModel},
+    model_names::Vector{<:AbstractString})`: Construct a `StackObservationModels` object with a vector of observation models and a
+    vector of model names.
+- `StackObservationModels(models::Vector{<:AbstractTuringObservationModel})`:
+    Construct a `StackObservationModels` object with a vector of observation models. The model names are automatically generated from the observation model types.
+- `StackObservationModels(models::NamedTuple{names, T})`: Construct a
+    `StackObservationModels` object with a named tuple of observation models. The model names are automatically generated from the keys of the named tuple.
+
+## Example
+
+```julia
+using EpiAware, Turing
+
+obs = StackObservationModels(
+    (cases = PoissonError(), deaths = NegativeBinomialError())
+)
+y_t = (cases = missing, deaths = missing)
+obs_model = generate_observations(obs, y_t, fill(10, 10))
+rand(obs_model)
+samples = sample(obs_model, Prior(), 100; progress = false)
+
+cases_y_t = group(samples, \"cases.y_t\")
```

Comment:
```suggestion
cases_y_t = group(samples, "cases.y_t")
```

Reply:
Is this a valid docstring? It wasn't rendering for me due to the external "" wrap from @doc raw or was that a late night delusion?

Reply:
Ah right! No you're good, I forgot about the string escape